lib/src/configuration.cc

```diff
@@ -683,6 +683,22 @@ void Configuration::validateAndNormalizeOtherSettings() {
         }
 
         auto pid_dir = fs::path(pid_file).parent_path().string();
+
+#ifdef __sun
```

Comment:
Should we enable this logic everywhere? E.g. someone might configure a linux distro like this as well.